index.ts

```diff
@@ -233,9 +243,17 @@ export default function viteNodeCGPlugin(pluginConfig: PluginConfig): Plugin {
             }
         },
 
-        configResolved(resolvedConfig: ResolvedConfig) {
+        async configResolved(resolvedConfig: ResolvedConfig) {
             // Capture resolved config for use in injectAssets
             config = resolvedConfig
+            // Check to see if one of the plugins is vite:react-refresh
+            if (resolvedConfig.plugins.find((plugin) => plugin.name === 'vite:react-refresh')) {
+                // If it is, import it and get the preamble code from it if possible
+                reactPreamble = (await import('@vitejs/plugin-react'))?.default?.preambleCode
```

Comment:
this part should handle when `@vitejs/plugin-react` is not installed, in which case it would throw an error